### DISCUSSION TITLE
Send to next Space

### DISCUSSION THREAD

**Discussion by kristiankruse** [rxhanson/RectanglePro-Community#8] [Ideas]
I don't use multiple monitors but Spaces—is it possible to have "send to next/prev Space" or does macOS limits that functionality?
Thank you.

**Comment by rxhanson** [comment]
This is a macOS limitation, unfortunately. There's no public API to manipulate spaces, and the only ways to do it are not the most robust. I haven't been inclined to add it in since I can't be confident that it will work for everyone.

**Reply by 9mm** [reply]
do you know how other window managers do it like size-up?

**Reply by rxhanson** [reply]
I don't know how SizeUp did it (but it was rather flaky for me in SizeUp, anyway). You can look at how Amethyst does it since it's open source. It's been a while since I looked, but I think Amethyst was simulating a user clicking on a window's title bar and dragging it up into Mission Control.

**Comment by kristiankruse** [comment]
Makes sense and thought so. Would love it though :)

**Reply by joernroeder** [reply]
same. what a bummer that this API does not exists

**Comment by rxhanson** [comment]
In the most recent release, 3.0.12, I added in "move to next/prev space" keyboard shortcuts.

**Reply by philoupd** [reply]
Would it be possible to add "move to next/prev space" as a window throw and long throw option. So far that's the only thing I am missing.

**Reply by rxhanson** [reply]
I didn't enable it in my first cut because it requires moving the cursor programmatically, which had some inherent flakiness with window throw, being based on cursor movement. It might be something that I can figure out, and it's on my backlog to take another look.

**Reply by BrentEubanks** [reply]
New to Rectangle Pro, I'm trying things out and trying to figure out if this is the solution for me.

I was looking for this functionality (move a window to the next/previous adjacent Space, ideally with a throw).  I found Next Space/Previous Space options in Settings and turned them on, but this leaves me with a few questions:

These options do not appear in the Throw menu and they also do not appear to have keyboard shortcuts.  The only way to trigger them seems to be from the menu bar dropdown.  Am I overlooking something?
<img width="340" alt="image" src="https://github.com/user-attachments/assets/fdc6b226-b6d9-4907-bfad-2b699ed054db" />

Is the functionality of these shortcuts still limited to just Chrome, Notion and Slack (per thread below)?  Are there plans to make it more universal?

Have you had any luck and/or do you have any update (one way or the other) in terms of being able to make this a throw gesture?

Thanks!

**Reply by BrentEubanks** [reply]
> These options do not appear in the Throw menu and they also do not appear to have keyboard shortcuts.

So I figured out how to assign keyboard shortcuts :)
I hadn't seen them on the list previously (because those commands were not enabled by default) but they're available to be defined now, so that answers that question.

My other question is still open though: Is there any way to send a window to an adjacent space without simultaneously moving focus to that Space at the same time?  I suspect the answer is still "no" but hoping to confirm.

**Reply by rxhanson** [reply]
Indeed, there is no (supported) way to send a window to an adjacent space without moving focus to that space. 

I'll take another look at throwing to another space.

**Comment by BigGillyStyle** [comment]
> In the most recent release, 3.0.12, I added in "move to next/prev space" keyboard shortcuts.

I just downloaded Rectangle Pro. When I created shortcuts for "next/prev space", it switched spaces for me but did not move the app that had focus. Am I potentially using the product wrong?

MacOS: Ventura 13.6
App I tried to move: Chrome 118.0.5993.117
Monitors: Macbook + 2 external 32" 4k

**Reply by rxhanson** [reply]
Does this happen for all apps or just certain ones?

**Reply by BigGillyStyle** [reply]
I performed an experiment this morning with many of the apps I use at work.  Here are the results of trying to move apps to next/prev space:

Success:
- Zoom
- MS Teams
- Excel

Failure:
- Chrome
- Notion
- Slack

**Reply by rxhanson** [reply]
Thanks. I'll see what I can do for the next release.

**Reply by rxhanson** [reply]
Looks like I never updated this thread, but I did put in a fix for this a while back and it should be working for Chrome, Notion, & Slack.
